addons/website_sale/models/product.py

```diff
@@ -96,15 +96,18 @@ def _is_available_on_website(self, website_id):
         Change in this method should be reflected in `_get_website_pricelists_domain`.
         """
         self.ensure_one()
-        return self.website_id.id == website_id or (not self.website_id and (self.selectable or self.sudo().code))
+        company_id = self.env["website"].browse(website_id).company_id.id
```

Comment:
> But since I don't personally like to have difficult-to-read conditions, and I think that one was already hard enough to not add another layer.

@rdeodoo  Same here, I have updated the commit.